pkg/operator/controller/controller.go

```diff
@@ -212,24 +214,32 @@ func haTypeForInfra(infraConfig *configv1.Infrastructure) ingressv1alpha1.Cluste
 	return ingressv1alpha1.UserDefinedClusterIngressHA
 }
 
-// effectiveHighAvailability returns the HA configuration as derived from the
-// given clusteringress and infrastructure config.
-func effectiveHighAvailability(ci *ingressv1alpha1.ClusterIngress, infraConfig *configv1.Infrastructure) (ha ingressv1alpha1.ClusterIngressHighAvailability) {
+// enforceEffectiveHighAvailability determines the effective HA configuration
+// for the given clusteringress and infrastructure config and publishes it to
+// the clusteringress's status.
+func (r *reconciler) enforceEffectiveHighAvailability(ci *ingressv1alpha1.ClusterIngress, infraConfig *configv1.Infrastructure) error {
+	// The clusteringress's HA configuration is immutable, so if we have
+	// published an HA type status, we must continue using it.
+	if len(ci.Status.HighAvailability.Type) > 0 {
```

Comment:
Do we need something like `UnknownClusterIngressHAType = ""` that we can use here? I'm not sure whether relying on the underlying type's zero value smells.

Reply:
length check looks fine to me for now